test/ScriptExecutorTests/ScriptExecutorTests.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using System.Runtime.InteropServices;
+using Xunit;
+using FluentAssertions;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.Cli.Utils.ScriptExecutorTests
+{
+    public class ScriptExecutorTests : TestBase
+    {
+        private static readonly string s_testProjectRoot = Path.Combine(AppContext.BaseDirectory, "TestAssets/TestProjects");
+
+        private TempDirectory _root;
+
+        public ScriptExecutorTests()
+        {
+            _root = Temp.CreateDirectory();
+        }
+
+        [Fact]
+        public void Test_Project_Local_Script_is_Resolved()
+        {
+            var sourceTestProjectPath = Path.Combine(s_testProjectRoot, "TestApp");
+            var binTestProjectPath = _root.CopyDirectory(sourceTestProjectPath).Path;
+
+            var project = ProjectContext.Create(binTestProjectPath, NuGetFramework.Parse("dnxcore50")).ProjectFile;
+
+            CreateTestFile("some.script", binTestProjectPath);
+            var scriptCommandLine = "some.script";
+
+            var command = ScriptExecutor.CreateCommandForScript(project, scriptCommandLine, new Dictionary<string, string>());
+
+            command.Should().NotBeNull();
+            command.ResolutionStrategy.Should().Be(CommandResolutionStrategy.ProjectLocal);
+        }
+        
+        [Fact]
+        public void Test_Nonexistent_Project_Local_Script_is_not_Resolved()
+        {
+            var sourceTestProjectPath = Path.Combine(s_testProjectRoot, "TestApp");
+            var binTestProjectPath = _root.CopyDirectory(sourceTestProjectPath).Path;
+
+            var project = ProjectContext.Create(binTestProjectPath, NuGetFramework.Parse("dnxcore50")).ProjectFile;
+
+            var scriptCommandLine = "nonexistent.script";
+
+            Action action = () => ScriptExecutor.CreateCommandForScript(project, scriptCommandLine, new Dictionary<string, string>());
+            action.ShouldThrow<CommandUnknownException>();
+        }
+        
+        [Fact]
+        public void Test_Extension_Inference_in_Resolution_for_Project_Local_Scripts()
```

Comment:
What do you expect here? Can you add the expected behavior in the test title like you did for the other ones.